src/s2e-kvm-vcpu-arm.h

```diff
@@ -0,0 +1,173 @@
+///
+/// Copyright (C) 2019, Cyberhaven
+/// All rights reserved.
+///
+/// Licensed under the Cyberhaven Research License Agreement.
+///
+
+#ifndef S2E_KVM_VCPU_H
```

Comment:
This file looks duplicated from x86. It would make sense to reuse common parts.
E.g., you could have a general abstract VCPU class that handles things that are common between architectures and then have a derived X86/ARM implementation.

Reply:
Another approach would just be to use #ifdefs like you did for the corresponding cpp file.